cmd/orchestrate/orchestrate.go

```diff
@@ -157,33 +159,51 @@ func processQueue(mq *broker.AMQPBroker, queue string, routingKey string, conf *
 		switch routingKey {
 		case conf.Orchestrator.QueueAccession:
 			publishMsg, publishType = finalizeMessage(delivered.Body, conf)
+			err = validateMsg(&delivered, mq, routingKey, durable, routingSchema, publishMsg, publishType)
+			if err != nil {
+				log.Errorf("Validation of outgoing message failed, error: %v", err)
+				if err := delivered.Nack(false, true); err != nil {
+					log.Errorf("failed to nack message for reason: %v", err)
+				}
+
+				continue
+			}
 		case conf.Orchestrator.QueueIngest:
 			publishMsg, publishType = ingestMessage(delivered.Body)
+			err = validateMsg(&delivered, mq, routingKey, durable, routingSchema, publishMsg, publishType)
+			if err != nil {
+				log.Errorf("Validation of outgoing message failed, error: %v", err)
+				if err := delivered.Nack(false, true); err != nil {
+					log.Errorf("failed to nack message for reason: %v", err)
+				}
+
+				continue
+			}
 		case conf.Orchestrator.QueueMapping:
 			publishMsg, publishType = mappingMessage(delivered.Body, conf)
-		}
-
-		err = mq.ValidateJSON(&delivered, routingSchema, publishMsg, publishType)
-
-		if err != nil {
-			log.Errorf("Validation of outgoing message failed, error: %v", err)
-			if err := delivered.Nack(false, true); err != nil {
-				log.Errorf("failed to nack message for reason: %v", err)
+			err = validateMsg(&delivered, mq, routingKey, durable, routingSchema, publishMsg, publishType)
+			if err != nil {
+				log.Errorf("Validation of outgoing message failed, error: %v", err)
+				if err := delivered.Nack(false, true); err != nil {
+					log.Errorf("failed to nack message for reason: %v", err)
+				}
+
+				continue
+			}
+			// let us wait a minute before sending the release message
+			time.Sleep(1 * time.Minute)
```

Comment:
Why do we wait a minute? Should it be configurable?

Reply:
just as default, i can make it configurable